### DISCUSSION TITLE
How do I force header files of dependencies of static library to be inherited by dependants?

### DISCUSSION THREAD

**Discussion by smac89** [xmake-io/xmake#5961] [Q&A] [answered]
Given the following scenario:

```lua
target("foo")
    set_kind("static")
    add_packages("poco")
    add_files("foo/src/**.cpp")
    add_includedirs("include", {public = true})

target("bar")
    set_kind("binary")
    add_deps("foo")
    -- add_packages("poco")
    add_files("bar/src/**.cpp")
```

I noticed that without the `add_packages("poco")` line in `bar` target, compilation doesn't work when a header file from `poco` package is encountered.

Is there a way to specify that the `poco` header files (from `foo`), should also be inherited by `bar`, without having to specify `add_packages("poco")`, in `bar`?

**Comment by waruqi** [comment, marked as answer]
```
add_packages("poco", {public = true})
```
